### PR TITLE
Add API endpoint for currencies

### PR DESCRIPTION
**Pull request by shal**

### CODE REVIEW

File: app/api/api_v2/currencies.rb [hunk 1]
```diff
@@ -15,5 +15,18 @@ class Currencies < Grape::API
       end
     end
 
+    desc 'Get list of currencies'
+    params do
+      optional :type, type: String, values: %w[fiat coin], desc: 'Type of currency. Available values: coin, fiat'
+    end
+    get '/currencies' do
+      if params[:type].nil?
```
Comment (ysv) [inline app/api/api_v2/currencies.rb]: `params[:type].blank?`

File: app/api/api_v2/entities/currency.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      expose :supports_cash_addr_format, if: { type: 'coin'}
+      expose :supports_hd_protocol, if: { type: 'coin'}
+      expose :allow_multiple_deposit_addresses, if: { type: 'coin'}
+
+      expose :wallet_url_template, if: { type: 'coin'}, documentation: 'URL template for coin wallet'
+      expose :transaction_url_template, if: { type: 'coin'}, documentation: 'Transaction URL template for coin wallet'
+
+      expose :base_factor, documentation: 'Currency base factor'
+      expose :precision, documentation: 'Currency precision'
+    end
```
Comment (ysv) [inline app/api/api_v2/entities/currency.rb]: expose only: 
id, symbol, type, deposit_fee, withdraw_fee, quick_withdraw_limit, deposit_confirmations, allow_multiple_deposit_addresses, base_factor, precision

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

File: spec/api/api_v2/currencies_spec.rb [hunk 3]
```diff
@@ -67,10 +66,67 @@
 
     after { KlineDB.redis.flushall }
 
+    RSpec::Matchers.define :have_trade_structure do
+      match { |x| not (x.dig(:price).nil? || x.dig(:volume).nil? || x.dig(:change).nil?) }
```
Comment (ysv) [inline spec/api/api_v2/currencies_spec.rb]:
```ruby
match { |x| x[:price].present? && x[:volume].present? && x[:change].present? }
```

**Review by ysv — CHANGES_REQUESTED** [review]

**Comment by ysv** [comment]
Please update doc https://github.com/rubykube/peatio/blob/master/docs/api/member_api_v2.md

**Comment by shal** [comment]
@ysv Travis CI will do this for us.
https://github.com/rubykube/peatio/blob/dc52324d4be87ab725056d922d5058b4ed60eda4/bin/bump#L35

**Comment by ysv** [comment]
@shal I don't think we run bump in master

**Comment by shal** [comment]
@ysv I think we do
https://github.com/rubykube/peatio/blob/dc52324d4be87ab725056d922d5058b4ed60eda4/ci/bump.rb#L103

**Comment by ysv** [comment]
@shal I don't think so https://github.com/rubykube/peatio/blob/dc52324d4be87ab725056d922d5058b4ed60eda4/ci/bump.rb#L227-L228

**Comment by shal** [comment]
@ysv you won :smile:

**Comment by shal** [comment]
@ysv I updated member API documentation according to this patch.

**Comment by shal** [comment]
@ysv Ready.

**Comment by ysv** [comment]
@shal looks good for me

**Comment by shal** [comment]
@ysv Please, push this PR into `wb1` branch to test it.

**Pull request merged**